src/community/label_propagation.c

```diff
@@ -213,163 +609,35 @@ igraph_error_t igraph_community_label_propagation(const igraph_t *graph,
+            VECTOR(*membership)[i] = i;
         }
     }
 
-    reversed_mode = IGRAPH_REVERSE_MODE(mode);
-
     /* From this point onwards we use 'fixed_copy' instead of 'fixed' */
+    IGRAPH_VECTOR_INIT_FINALLY(&label_counters, no_of_nodes);
 
-    /* Create an adjacency/incidence list representation for efficiency.
-     * For the unweighted case, the adjacency list is enough. For the
-     * weighted case, we need the incidence list */
-    if (weights) {
-        IGRAPH_CHECK(igraph_inclist_init(graph, &il, reversed_mode, IGRAPH_LOOPS_ONCE));
-        IGRAPH_FINALLY(igraph_inclist_destroy, &il);
-    } else {
-        IGRAPH_CHECK(igraph_adjlist_init(graph, &al, reversed_mode, IGRAPH_LOOPS_ONCE, IGRAPH_MULTIPLE));
-        IGRAPH_FINALLY(igraph_adjlist_destroy, &al);
-    }
-
-    /* Create storage space for counting distinct labels and dominant ones */
-    IGRAPH_VECTOR_INIT_FINALLY(&label_counters, no_of_nodes + 1);
-    IGRAPH_VECTOR_INT_INIT_FINALLY(&dominant_labels, 0);
-    IGRAPH_VECTOR_INT_INIT_FINALLY(&nonzero_labels, 0);
-    IGRAPH_CHECK(igraph_vector_int_reserve(&dominant_labels, 2));
-
-    /* Initialize node ordering vector with only the not fixed nodes */
-    if (fixed_copy) {
-        IGRAPH_VECTOR_INT_INIT_FINALLY(&node_order, no_of_not_fixed_nodes);
-        for (i = 0, j = 0; i < no_of_nodes; i++) {
-            if (!VECTOR(*fixed_copy)[i]) {
-                VECTOR(node_order)[j] = i;
-                j++;
-            }
-        }
-    } else {
-        IGRAPH_CHECK(igraph_vector_int_init_range(&node_order, 0, no_of_nodes));
-        IGRAPH_FINALLY(igraph_vector_int_destroy, &node_order);
-    }
-
-    /* There are two alternating types of iterations, one for changing labels and
-    the other one for checking the end condition - every vertex in the graph has
-    a label to which the maximum number of its neighbors belongs. If control_iteration
-    is true, we are just checking the end condition and not relabeling nodes. */
-    control_iteration = true;
-    running = true;
-    while (running) {
-        igraph_integer_t v1, num_neis;
-        igraph_real_t max_count;
-        igraph_vector_int_t *neis;
-        igraph_vector_int_t *ineis;
-        igraph_bool_t was_zero;
-
-        if (control_iteration) {
-            /* If we are in the control iteration, we expect in the beginning of
-            the iteration that all vertices meet the end condition, so 'running' is false.
-            If some of them does not, 'running' is set to true later in the code. */
-            running = false;
-        } else {
-            /* Shuffle the node ordering vector if we are in the label updating iteration */
-            igraph_vector_int_shuffle(&node_order);
-        }
-
-        RNG_BEGIN();
-        /* In the prescribed order, loop over the vertices and reassign labels */
-        for (i = 0; i < no_of_not_fixed_nodes; i++) {
-            v1 = VECTOR(node_order)[i];
+    switch(lpa_variant)
```

Comment:
Can you run this file through astyle and add an extra commit with the result of that formatting?

Reply:
Done, but barely made any changes.

Reply:
Done, but barely made a difference.

Reply:
> but barely made a difference

I believe this is because you did not use our astyle settings. Use the `.astylerc` in the top level directory of the repo. This should do it:

```
astyle --project=.astylerc src/community/label_propagation.c
```

Do you want me to do it and push to this branch (I hope I can manage to do that)?